src/_pytest/pytester_assertions.py

```diff
@@ -0,0 +1,56 @@
+"""Helper plugin for pytester; should not be loaded on its own."""
+# This plugin contains assertions used by pytester. pytester cannot
+# contain them itself, since it is imported by the `pytest` module,
+# hence cannot be subject to assertion rewriting, which requires a
+# module to not be already imported.
+from _pytest.pytester import HookRecorder
+from _pytest.pytester import RunResult
```

Comment:
I think it might be cleaner to avoid the cyclic import here, and have `assertoutcome` take the return value of `listoutcomes`, and assert_outcomes take the return value of `parseoutcomes`

Reply:
I wasn't really bothered with a cyclic import in this specific case, but I it is better to avoid -- will update.